<launch>
  <arg name="is_sim" default="$(optenv IS_SIM false)"/>
  <arg name="annotation_file" default="$(find cafe_concert)/annotations/dorothy.yaml"/>
  <arg name="annotation_namespace" default="/annotation"/>


  <arg     if="$(arg is_sim)" name="map_file" value="$(find cafe_concert)/maps/dorothy_cafe_new_polished.yaml"/>
  <arg unless="$(arg is_sim)" name="map_file" value="$(find cafe_concert)/maps/dorothy_cafe_new.yaml"/>

  <!-- Publish a dummy tf, as map viewer on Android app needs a tf reference
       relative to which show the map on screnn (logically we use /map) -->
  <node pkg="tf" type="static_transform_publisher" name="dummy_map_tf_broadcaster" args="0 0 0 0 0 0 map dummy_tf 10" />

  <!-- *************************** Maps *************************** -->
  <node pkg="map_server" type="map_server" name="map_server" args="$(arg map_file)">
    <param name="frame_id" value="/map"/>
    <remap from="map" to="/map"/>
    <remap from="map_metadata" to="/map_metadata"/>
    <remap from="static_map" to="/static_map"/>
  </node>

  <!-- *********************** Annotations ************************ -->
  <node pkg="simple_annotation_server" type="simple_server.py" name="annotations_manager">
    <param name="filename" value="$(arg annotation_file)"/>
    <remap from="ar_markers" to="marker_pose_list"/>
    <remap from="tables" to="table_pose_list"/>
    <remap from="walls" to="$(arg annotation_namespace)/walls"/>
    <remap from="columns" to="$(arg annotation_namespace)/columns"/>

    <remap from="save_columns" to="$(arg annotation_namespace)/save_columns"/>
    <remap from="save_tables" to="$(arg annotation_namespace)/save_tables"/>
    <remap from="save_walls" to="$(arg annotation_namespace)/save_walls"/>
    <remap from="save_ar_markers" to="$(arg annotation_namespace)/save_ar_markers"/>
  </node>

</launch>
